archivebox: refactor and fix

Jonasmadsen opened the pull request:
###### Motivation for this change
Archivebox didn't build due to conflicting dependency on django.
Possibly related to Issue https://github.com/NixOS/nixpkgs/issues/154950 that was closed by PR https://github.com/NixOS/nixpkgs/pull/154153 

###### Things done
Resolves conflict by overriding django dependency for all dependencies.
Updated commit message and using packageOverrides as requested by reviewers on previous now closed PR https://github.com/NixOS/nixpkgs/pull/156520

Basic functionality tested and working.
Nothing else should depend on archivebox.

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))

- [ ] Tested, as applicable:
  - [X] NixOS
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

dotlambda commented on pkgs/applications/misc/archivebox/default.nix:
```diff
@@ -1,27 +1,22 @@
+      django = super.django_3.overridePythonAttrs (old: rec {
+        pname = "Django";
+        version = "3.1.7";
+        src = fetchPypi {
+          inherit pname version;
+          sha256 = "sha256-Ms55Lum2oMu+w0ASPiKayfdl3/jCpK6SR6FLK6OjZac=";
+        };
```
```suggestion
        src = old.src.override {
          inherit version;
          sha256 = "sha256-Ms55Lum2oMu+w0ASPiKayfdl3/jCpK6SR6FLK6OjZac=";
        };
```

dotlambda commented on pkgs/applications/misc/archivebox/default.nix:
```diff
@@ -1,27 +1,22 @@
+  py = python3.override {
+    packageOverrides = self: super: {
+      django = super.django_3.overridePythonAttrs (old: rec {
+        pname = "Django";
```
```suggestion
        pname = "django";
```
https://github.com/NixOS/nixpkgs/pull/158710

dotlambda commented on pkgs/applications/misc/archivebox/default.nix:
```diff
@@ -1,27 +1,22 @@
 { lib
 , buildPythonApplication
+, python3
 , fetchPypi
```
```suggestion
, python3
```

dotlambda commented on pkgs/applications/misc/archivebox/default.nix:
```diff
@@ -1,27 +1,22 @@
+  };
 in
 
 buildPythonApplication rec {
```
```suggestion
python.pkgs.buildPythonApplication rec {
```

dotlambda commented on pkgs/applications/misc/archivebox/default.nix:
```diff
@@ -1,27 +1,22 @@
-    src = fetchPypi {
-      inherit pname version;
-      sha256 = "sha256-Ms55Lum2oMu+w0ASPiKayfdl3/jCpK6SR6FLK6OjZac=";
+  py = python3.override {
```
```suggestion
  python = python3.override {
```

dotlambda commented on pkgs/applications/misc/archivebox/default.nix:
```diff
@@ -33,10 +28,10 @@ buildPythonApplication rec {
   };
 
-  propagatedBuildInputs = [
+  propagatedBuildInputs = with py.pkgs; [
```
```suggestion
  propagatedBuildInputs = with python.pkgs; [
```

dotlambda reviewed (changes requested):
Please don't use `python3Packages.callPackage` in `all-packages.nix`.

Jonasmadsen commented:
@dotlambda accepted your changes

dotlambda commented on pkgs/applications/misc/archivebox/default.nix:
```diff
@@ -1,42 +1,35 @@
   version = "0.6.2";
 
-  src = fetchPypi {
+  src = python3.pkgs.fetchPypi {
```
```suggestion
  src = python.pkgs.fetchPypi {
```

siraben commented:
Result of `nixpkgs-review pr 158704` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>archivebox</li>
  </ul>
</details>

dotlambda commented:
> 1 package failed to build

@siraben Was that due to ipython, like for me on aarch64-linux?

siraben commented:
@dotlambda yes, that's the cause of the failure.

Commit:
archivebox: refactor and fix

Archivebox didnt build before but does now. 
Refactor with packageOverrides
Update pkgs/applications/misc/archivebox/default.nix

Co-authored-by: Robert Schütz <github@dotlambda.de>
Update pkgs/applications/misc/archivebox/default.nix

Co-authored-by: Robert Schütz <github@dotlambda.de>
Update pkgs/applications/misc/archivebox/default.nix

Co-authored-by: Robert Schütz <github@dotlambda.de>
Update pkgs/applications/misc/archivebox/default.nix

Co-authored-by: Robert Schütz <github@dotlambda.de>
Update pkgs/applications/misc/archivebox/default.nix

Co-authored-by: Robert Schütz <github@dotlambda.de>
Update pkgs/applications/misc/archivebox/default.nix

Co-authored-by: Robert Schütz <github@dotlambda.de>
fetchPypi didn't exist in scope


don't use python3Packages.callPackage


Update pkgs/applications/misc/archivebox/default.nix

Co-authored-by: Robert Schütz <github@dotlambda.de>
Update pkgs/applications/misc/archivebox/default.nix

Co-authored-by: Robert Schütz <github@dotlambda.de>
